### PR TITLE
[JIT] Fixed assertion for targetType and op1 type not being the same

### PR DESCRIPTION
**Pull request by TIHan**
Resolves this issue: https://github.com/dotnet/runtime/issues/80773

### CODE REVIEW

**Comment by TIHan** [comment]
@dotnet/jit-contrib this is ready - @tannergooding @kunalspathak

File: src/coreclr/jit/lowerxarch.cpp [hunk 1]
```diff
@@ -1180,6 +1180,13 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
 
             if (op1IsVectorZero && op2IsVectorZero)
             {
+                // We need to change op1's type to the return type of the node in this case.
+                // We have to do this because we are going to propagate the constant up.
+                op1->ChangeType(node->TypeGet());
+
+                // Ensure the upper values are zero by zero-initialization.
+                op1->AsVecCon()->gtSimd32Val = {};
```
Comment (BruceForstall) [inline src/coreclr/jit/lowerxarch.cpp:1188]: Maybe there should be a `GenTreeVecCon::SetZero()` function to encapsulate this, adjacent to the existing `IsZero()`?

**Review by BruceForstall — APPROVED** [review]

**Review by tannergooding — APPROVED** [review]

File: src/coreclr/jit/lowerxarch.cpp (hunk 1 above)
Comment (TIHan) [reply src/coreclr/jit/lowerxarch.cpp:1188]: Probably would be a good idea to do.

**Pull request merged**